lib/ansible/modules/source_control/gitlab_hook.py

```diff
@@ -0,0 +1,383 @@
+            validate_certs=dict(required=False, default=True, type='bool', aliases=['verify_ssl']),
+            login_user=dict(required=False, no_log=True),
+            login_password=dict(required=False, no_log=True),
+            login_token=dict(required=False, no_log=True),
+            group=dict(required=True),
+            project=dict(required=True),
+            url=dict(required=True),
+            push_events=dict(default=True, type='bool'),
+            pipeline_events=dict(default=True, type='bool'),
+            issues_events=dict(default=False, type='bool'),
+            note_events=dict(default=False, type='bool'),
+            merge_requests_events=dict(default=False, type='bool'),
+            tag_push_events=dict(default=False, type='bool'),
+            build_events=dict(default=False, type='bool'),
+            enable_ssl_verification=dict(default=True, type='bool'),
+            state=dict(default="present", choices=["present", 'absent']),
+        ),
+        supports_check_mode=True
+    )
+
+    if not HAS_GITLAB_PACKAGE:
+        module.fail_json(msg="Missing required gitlab module (check docs or install with: pip install python-gitlab")
+
+    server_url = module.params['server_url']
+    verify_ssl = module.params['validate_certs']
+    login_user = module.params['login_user']
+    login_password = module.params['login_password']
+    login_token = module.params['login_token']
+    project = module.params['project']
+    group = module.params['group']
+    hook_url = module.params['url']
+    push_events = module.params['push_events']
+    issues_events = module.params['issues_events']
+    note_events = module.params['note_events']
+    pipeline_events = module.params['pipeline_events']
+    merge_requests_events = module.params['merge_requests_events']
+    tag_push_events = module.params['tag_push_events']
+    build_events = module.params['build_events']
+    enable_ssl_verification = module.params['enable_ssl_verification']
+    state = module.params['state']
+    use_credentials = None
+
+    # Validate some credentials configuration parameters.
+    if login_user is not None and login_password is not None:
+        use_credentials = True
+    elif login_token is not None:
+        use_credentials = False
+    else:
+        module.fail_json(msg="No login credentials are given. Use login_user with login_password, or login_token")
+
+    if login_token and login_user:
+        module.fail_json(msg="You can either use 'login_token' or 'login_user' and 'login_password'")
+
+    try:
+        if use_credentials:
+            git = gitlab.Gitlab(server_url, email=login_user, password=login_password, ssl_verify=verify_ssl)
+            git.auth()
+        else:
+            git = gitlab.Gitlab(server_url, private_token=login_token, ssl_verify=verify_ssl)
+            git.auth()
+    except Exception:
+        e = get_exception()
+        module.fail_json(msg="Failed to connect to Gitlab server: %s " % e)
+
```

Comment:
Can you please add check if user is authenticated or not ? in past, there was a bug related to this, which was fixed by using [this](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/source_control/gitlab_user.py#L345-L349)